dvc/fs/dvc.py

```diff
@@ -162,7 +162,12 @@ def _get_key(self, path) -> Key:
 
     def _get_key_from_relative(self, path) -> Key:
         parts = self.path.relparts(path, self.root_marker)
-        if parts and parts[0] == os.curdir:
+        if parts and any(
+            [
+                parts[0] == os.curdir,
+                parts[0] == self.path.split(os.getcwd())[-1],
```

Comment:
Since this says `from_relative()`, I am a bit suspicious about the fix. I'll look into it and try to see what the actual problem is.